pkg/ticker/ticker.go

```diff
@@ -4,25 +4,120 @@ import (
 	"time"
 
 	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/configurator"
 	"github.com/openservicemesh/osm/pkg/kubernetes/events"
 	"github.com/openservicemesh/osm/pkg/logger"
 )
 
+const (
+	// Value used to identify ticker stop (0)
+	stopTickerDuration    = time.Duration(0)
+	// Minimum allowed ticker duration
+	minimumTickerDuration = time.Duration(1 * time.Minute)
+)
+
+// ResyncTicker contains the stop configuration for the ticker routines
+type ResyncTicker struct {
+	stopTickerRoutine chan struct{}
+	stopConfigRoutine chan struct{}
+}
+
 var (
 	log = logger.New("ticker")
+
+	// Local reference to global ticker
+	rTicker *ResyncTicker = nil
 )
 
 // InitTicker initializes a global ticker that is configured via
 // pubsub, and triggers global proxy updates also through pubsub.
 // Upon this function return, the ticker is guaranteed to be started
 // and ready to receive new events.
-func InitTicker() {
+func InitTicker(c configurator.Configurator) *ResyncTicker {
+	if rTicker != nil {
+		return rTicker
+	}
+
+	// Start config resync ticker routine
 	doneInit := make(chan struct{})
-	go ticker(doneInit)
+	stopTicker := make(chan struct{})
+	go ticker(doneInit, stopTicker)
 	<-doneInit
+
+	// Start config listener
+	doneInit = make(chan struct{})
+	stopConfig := make(chan struct{})
+	go tickerConfigListener(c, doneInit, stopConfig)
+	<-doneInit
+
+	rTicker = &ResyncTicker{
+		stopTickerRoutine: stopTicker,
+		stopConfigRoutine: stopConfig,
+	}
+	return rTicker
+}
+
+// Returns if a given duration is valid
+func validTickerDuration(d time.Duration) bool {
```

Comment:
I think you didn't see, but there's a difference between a 0 period and a valid ticker.
I added a minimum threshold for ticker to tick (1m) to avoid users to shoot themselves in the foot.

Reply:
I see what you mean, taking your request to change the decision factor to start or stop based on the threshold.